pkg/api/query/v1.go

```diff
@@ -657,13 +657,13 @@ func (qapi *QueryAPI) labelNames(r *http.Request) (interface{}, []error, *api.Ap
 		return nil, nil, apiErr
 	}
 
-	var matcherSets [][]*labels.Matcher
+	var matcherSets []*labels.Matcher
```

Comment:
The behavior changes here. We need to support multiple matchers, same as the label values API.